packages/cli/src/tokens/build/types.ts

```diff
@@ -0,0 +1,37 @@
+import type SD from 'style-dictionary/types';
+import type { StyleDictionaryConfigs } from './configs';
+
+/**
+ * A multi-dimensional theme is a concrete permutation of the possible theme dimensions
+ */
+export type ThemePermutation = {
+  mode: string;
+  semantic: string;
+  size: string;
+  theme: string;
+  typography: string;
+};
+
+export type ThemeDimension = keyof ThemePermutation;
+
+export type IsCalculatedToken = (token: SD.TransformedToken, options?: SD.Config) => boolean;
+
+export type GetSdConfigOptions = {
+  outPath: string;
+  tokensDir: string;
+};
+
+export type BuildConfig = {
+  /** Optional name of the build config - only used in the console output */
+  name?: string;
+  /** Style Dictionary configuration id */
+  sdConfig: StyleDictionaryConfigs;
```

Comment:
Yes, I think the jsdoc-comment and type is explicit enough that this is Style Dictionary

Reply:
Although, the main reason for adding `sd` was that otherwise you get something like this, and I feel like the word `config` is now overloaded to mean two different things

```ts
const storefrontConfig = {
    name: 'Storefront preview tokens',
    config: 'typescriptTokens',
    dimensions: ['mode'],
    options: { outPath: path.resolve('../../apps/storefront/tokens') },
}

Reply:
I changed the field to `getConfig` and used the GetStyleDictionaryConfig functions directly instead of by name -- this also makes it more flexible/extensible, not sure what I was thinking was the benefit of getting the config by key 😅

Reply:
Nice!